### PR TITLE
[#187364750] A user should be able to search a product 

### PR DESCRIPTION
**Pull request by Bananayosostene**
**What does this PR do?**
This PR implements the feature allowing users to search for products. Users can now search by product name, price range, category, and a combination of these criteria using an AND operation.

**Description of Task to be completed?**
The task involves creating a search functionality for products on the e-commerce platform. Users, particularly buyers, need the ability to search for products based on various criteria such as name, price range, and category. The search can also combine multiple criteria to refine the results further. This feature enhances the user experience by making it easier for buyers to find products that meet their specific needs.

**How should this be manually tested?**
To manually test this feature, navigate to the search bar on the product listing page. Enter a product name, select a price range, and/or choose a category to search within. Verify that the results displayed match the search criteria provided. Additionally, test combinations of these criteria to ensure the AND operation is functioning correctly. For example, search for products within a certain price range and category to see if the results accurately reflect both conditions.

**Any background context I want to provide?**
Before this feature, users could not refine their searches effectively, leading to a potentially frustrating shopping experience. Implementing this search functionality addresses this issue by providing a more intuitive and efficient way to find products. This change aims to enhance overall user satisfaction and engagement on the platform.

**Relevant pivotal tracker/Trello stories**

**Video**
https://www.loom.com/share/8a231cc5f851474c950fd68ba37bf0ce

Questions:
If you have any questions or need further clarification about this feature, please feel free to ask.

### CODE REVIEW

File: src/controllers/productController.ts
Comment (Yvan-David) [inline src/controllers/productController.ts:1]: they are many changes on this file while you only worked on one feature please revert them to reduce conflicts

**Review by Yvan-David — CHANGES_REQUESTED** [review]

File: src/controllers/productController.ts
Comment (Yvan-David) [inline src/controllers/productController.ts:1]: resolve conflicts

**Review by Yvan-David — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -31,7 +31,6 @@ env:
   CLIENT_SECRET: ${{ secrets.CLIENT_SECRET}}
   CALLBACK_URL: ${{ secrets.CALLBACK_URL}}
 
-
```
Comment (Yvan-David) [inline .github/workflows/ci.yml]: remove this file
as a change by adding that space you deleted

**Review by Furaha-Justine — APPROVED** [review]
Good job

**Review by Yvan-David — APPROVED** [review]

**Review by munezerobagira — CHANGES_REQUESTED** [review]
can you remove  logs

**Review by munezerobagira — APPROVED** [review]

**Pull request merged**